ACA-Py / AFJ runset test failing on CredentialProposal message processing by ACA-Py (Acme)

[Failing Test Details](https://allure.vonx.io/api/allure-docker-service/projects/acapy-b-javascript/reports/498/index.html#testresult/ad429a23b9e9e77a) (link will last only 10 days). Per the test results, this test was passing up until June 20th.

Command to run: `./manage run -d acapy-main -b javascript -t AIP20 -t RFC0160 -t RFC0592 -t Schema_DriversLicense_v2 -t CredFormat_Indy -t RFC0453 -t T001-RFC0453 -t AcceptanceTest`

The problem is that Bob (AFJ) sends a CredentialProposal message, and ACA-Py seems to blow up trying to process it, with the error:

```
AssertionError: resp_status 500 is not 200; {"name":"ClassValidationError","message":"V1CredentialPreview: Failed to validate class.\nAn instance of V1CredentialPreview has failed the validation:\n - property type has failed the following constraints: isValidMessageType \n","status":500,"errors":[],"stack":"ClassValidationError: V1CredentialPreview: Failed to validate class.\nAn instance of V1CredentialPreview has failed the validation:\n - property type has failed the following constraints: isValidMessageType \n\n    at Function.validateSync (/src/node_modules/@aries-framework/core/src/utils/MessageValidator.ts:19:13)\n    at Function.fromJSON (/src/node_modules/@aries-framework/core/src/utils/JsonTransformer.ts:35:22)\n    at IssueCredentialV2Controller.<anonymous> (/src/src/controllers/IssueCredentialV2Controller.ts:71:37)\n    at Generator.next (<anonymous>)\n    at /src/node_modules/tslib/tslib.js:118:75\n    at new Promise (<anonymous>)\n    at Object.__awaiter (/src/node_modules/tslib/tslib.js:114:16)\n    at IssueCredentialV2Controller.$ctx [as sendProposal] (/src/src/controllers/IssueCredentialV2Controller.ts:44:24)\n    at /src/node_modules/@tsed/platform-params/src/builder/PlatformParams.ts:88:39\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"}
```

In reviewing the logs from a local run, it looks to me like the error is occurring on [this line](https://github.com/hyperledger/aries-cloudagent-python/blob/47dbb28c39bfd3f02da178c24aa8de4edee92f31/aries_cloudagent/protocols/issue_credential/v2_0/formats/indy/handler.py#L220) in ACA-Py create offer -- after getting the SchemaID from the proposal, getting the Schema from the ledger and then trying to get the list of attributes from the schema.  

`File "/usr/local/lib/python3.7/site-packages/aries_cloudagent/protocols/issue_credential/v2_0/formats/indy/handler.py", line 222, in create_offer   attr for attr in cred_proposal_message.credential_preview.attr_dict()`

Not sure that makes a lot of sense.  I used the logs to find the `Proposal` message received and it's attachment and found both looked correctly formatted, and then checked the ledger to confirm that all of the IDs in the message referenced objects found on the ledger.

--- a/aries-backchannels/javascript/server/src/controllers/IssueCredentialV2Controller.ts
+++ b/aries-backchannels/javascript/server/src/controllers/IssueCredentialV2Controller.ts
@@ -7,6 +7,7 @@ import {
   CredentialStateChangedEvent,
   JsonTransformer,
   V1CredentialPreview,
+  V2CredentialPreview,
 } from '@aries-framework/core'
 import { CredentialUtils } from '../utils/CredentialUtils'
 import { filter, firstValueFrom, ReplaySubject, timeout } from 'rxjs'
@@ -68,7 +69,9 @@ export class IssueCredentialV2Controller extends BaseController {
       credential_preview: any
     }
   ) {
-    const preview = JsonTransformer.fromJSON(data.credential_preview, V1CredentialPreview)
+    // Disable validation. the @type sent by AATH is 'issue-credential/2.0/credential-preview'
+    // which is not a valid message type.
+    const preview = JsonTransformer.fromJSON(data.credential_preview, V2CredentialPreview, { validate: false })
     const connection = await ConnectionUtils.getConnectionByConnectionIdOrOutOfBandId(this.agent, data.connection_id)
     const credentialRecord = await this.agent.credentials.proposeCredential({
       connectionId: connection.id,
@@ -123,7 +126,9 @@ export class IssueCredentialV2Controller extends BaseController {
       })
       return this.mapCredential(credentialRecord)
     } else if (data) {
-      const preview = JsonTransformer.fromJSON(data.credential_preview, V1CredentialPreview)
+      // Disable validation. the @type sent by AATH is 'issue-credential/2.0/credential-preview'
+      // which is not a valid message type.
+      const preview = JsonTransformer.fromJSON(data.credential_preview, V2CredentialPreview, { validate: false })
       const connection = await ConnectionUtils.getConnectionByConnectionIdOrOutOfBandId(this.agent, data.connection_id)
 
       if (!data.filter.indy.cred_def_id) {

--- a/aries-backchannels/javascript/server/src/controllers/MediationController.ts
+++ b/aries-backchannels/javascript/server/src/controllers/MediationController.ts
@@ -10,6 +10,7 @@ import {
 import { TestHarnessConfig } from '../TestHarnessConfig'
 import { BaseController } from '../BaseController'
 import { filter, firstValueFrom, ReplaySubject, timeout } from 'rxjs'
+import { ConnectionUtils } from '../utils/ConnectionUtils'
 
 @Controller('/agent/command/mediation')
 export class MediationController extends BaseController {
@@ -39,7 +40,8 @@ export class MediationController extends BaseController {
 
   @Get('/:connectionId')
   async getMediationRecordByConnectionId(@PathParams('connectionId') connectionId: string) {
-    const mediationRecord = await this.agent.mediationRecipient.findByConnectionId(connectionId)
+    const connection = await ConnectionUtils.getConnectionByConnectionIdOrOutOfBandId(this.agent, connectionId)
+    const mediationRecord = await this.agent.mediationRecipient.findByConnectionId(connection.id)
 
     if (!mediationRecord) {
       throw new NotFound(`mediation record for connectionId "${connectionId}" not found.`)
@@ -50,11 +52,7 @@ export class MediationController extends BaseController {
 
   @Post('/send-request')
   async sendMediationRequest(@BodyParams('id') connectionId: string) {
-    const connection = await this.agent.connections.findById(connectionId)
-
-    if (!connection) {
-      throw new NotFound(`connection record with id "${connectionId}" not found.`)
-    }
+    const connection = await ConnectionUtils.getConnectionByConnectionIdOrOutOfBandId(this.agent, connectionId)
 
     const mediationRecord = await this.agent.mediationRecipient.requestMediation(connection)
 
@@ -63,9 +61,10 @@ export class MediationController extends BaseController {
 
   @Post('/send-grant')
   async sendMediationGrant(@BodyParams('id') connectionId: string) {
+    const connection = await ConnectionUtils.getConnectionByConnectionIdOrOutOfBandId(this.agent, connectionId)
     let {
       payload: { mediationRecord },
-    } = await this.waitForState(connectionId, MediationState.Granted)
+    } = await this.waitForState(connection.id, MediationState.Granted)
 
     // Auto accept is enabled, so we're not granting the mediation explicitly
     return this.mapMediationRecord(mediationRecord)

--- a/aries-backchannels/javascript/server/src/controllers/PresentProofController.ts
+++ b/aries-backchannels/javascript/server/src/controllers/PresentProofController.ts
@@ -100,7 +100,10 @@ export class PresentProofController extends BaseController {
       }
     }
   ) {
-    const proofRequest = JsonTransformer.fromJSON(data.presentation_request.proof_request.data, ProofRequest)
+    // Do not validate, we only need a few properties from the proof request
+    const proofRequest = JsonTransformer.fromJSON(data.presentation_request.proof_request.data, ProofRequest, {
+      validate: false,
+    })
 
     const connection = await ConnectionUtils.getConnectionByConnectionIdOrOutOfBandId(this.agent, data.connection_id)
 

